### DISCUSSION TITLE
Passing data-* attributes to the Table, rows and cells

### DISCUSSION THREAD

**Discussion by darktasevski** [KevinVandy/material-react-table#825] [Ideas]
It looks like passing `data` attributes is not currently supported. I usually use these for testing purposes, setting `data-test-id` attributes to rows and cells, but I'm sure there are scenarios where passing `data-` attributes would be helpful.

MUI components have no problems with accepting `data attributes and I can easily pass them, for example:
```tsx
<Paper data-test-id="some_test_selector">
	...
```

But when I try to pass the `data-test-id` to the table configuration I'm getting Ts errors:

![CleanShot 2023-11-17 at 13 52 06](https://github.com/KevinVandy/material-react-table/assets/25148427/a1845873-fb0d-47fc-83bf-b499cfa095c3)

**Comment by KevinVandy** [comment]
It's supported, you just have to use ts ignore. This is a well known problem with the react types
